src/braket/aws/aws_quantum_task_batch.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2019-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+import time
+from concurrent.futures.thread import ThreadPoolExecutor
+from typing import List, Set, Union
+
+from braket.annealing import Problem
+from braket.aws.aws_quantum_task import AwsQuantumTask
+from braket.aws.aws_session import AwsSession
+from braket.circuits import Circuit
+
+
+class AwsQuantumTaskBatch:
+    """Executes and a batch of quantum tasks in parallel.
+
+    Using this class can yield vast speedups over executing tasks sequentially,
+    and is particularly useful for computations that can be parallelized,
+    such as calculating quantum gradients or statistics of terms in a Hamiltonian.
+
+    Note: there is no benefit to using this method with QPUs outside of their execution windows,
+    since results will not be available until the window opens.
+    """
+
+    MAX_WORKERS = 100
```

Comment:
Should we try to feed this into the max_pool_size for the configs for the s3 and braket client as well?